src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+    * non-empty in the provided delta. Broadcast fields are broadcasted here as well, as needed.
+    *
+    * Note: this is only called by Spark threads.
+    */
+  private def updateTransformContext(delta: Delta, sparkContext: SparkContext): Unit = {
+    if (transformContext == null) {
+      transformContext = TransformContext()
+    }
+
+    transformContext = transformContext.copy(
+      siteSettings = delta.siteSettings.getOrElse(transformContext.siteSettings),
+      langToWatchlist = delta.langToWatchlist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.langToWatchlist
+      },
+      blacklist = delta.blacklist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.blacklist
+      },
+      locationsExtractorFactory = delta.locationsExtractorFactory match {
+        case Some(factory) => sparkContext.broadcast(factory)
+        case None => transformContext.locationsExtractorFactory
+      },
+      imageAnalyzer = delta.imageAnalyzer.getOrElse(transformContext.imageAnalyzer),
+      languageDetector = delta.languageDetector.getOrElse(transformContext.languageDetector),
+      sentimentDetectorAuth = delta.sentimentDetectorAuth.getOrElse(transformContext.sentimentDetectorAuth)
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided site settings.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithSettings(siteSettings: SiteSettings, delta: Delta = Delta()): Delta = {
+    // Note that parameters are call-by-name
+    def updatedField[T](isDirty: => Boolean, newVal: => T): Option[T] = {
+      // Invoke 'isDirty' only if transform context is not null
+      if (transformContext == null || isDirty)
+        Some(newVal)
+      else
+        None
+    }
+
+    delta.copy(
+      siteSettings = Some(siteSettings),
+      locationsExtractorFactory = updatedField(
+        siteSettings.geofence != transformContext.siteSettings.geofence,
+        new LocationsExtractorFactory(featureServiceClient, siteSettings.geofence).buildLookup()
+      ),
+      imageAnalyzer = updatedField(
+        siteSettings.cogVisionSvcToken != transformContext.siteSettings.cogVisionSvcToken,
+        new ImageAnalyzer(ImageAnalysisAuth(siteSettings.cogVisionSvcToken), featureServiceClient)
+      ),
+      languageDetector = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        new LanguageDetector(LanguageDetectorAuth(siteSettings.translationSvcToken))
+      ),
+      sentimentDetectorAuth = updatedField(
+        siteSettings.translationSvcToken != transformContext.siteSettings.translationSvcToken,
+        SentimentDetectorAuth(siteSettings.translationSvcToken)
+      )
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided watchlist.
```

Comment:
Well actually, if we end up updating per-word, this function can copy the existing `langToWatchlist`, adding/removing a word.